core/src/main/java/org/elasticsearch/discovery/Discovery.java

```diff
@@ -87,4 +88,10 @@ public FailedToCommitClusterStateException(String msg, Throwable cause, Object..
             super(msg, cause, args);
         }
     }
+
+    /**
+     * @return stats about the discovery
+     */
+    PendingClusterStateStats stats();
```

Comment:
Re the cat api - I'm fine if this is not included in them btw - this is a super expert stats.